cmd/relayproxy/config/config.go

```diff
@@ -91,10 +92,15 @@ func New(flagSet *pflag.FlagSet, log *zap.Logger, version string) (*Config, erro
 			log.Error("error loading file", zap.Error(errBindFile))
 		}
 	}
-
+	configMap := k.Raw()
 	// Map environment variables
-	_ = k.Load(env.Provider("", ".", func(s string) string {
-		return strings.ReplaceAll(strings.ToLower(s), "_", ".")
+	_ = k.Load(env.ProviderWithValue("", ".", func(s string, v string) (string, interface{}) {
+		if strings.HasPrefix(s, "RETRIEVERS") || strings.HasPrefix(s, "NOTIFIERS") {
+			err := loadArrayEnv(s, v, configMap)
+			log.Error("config: error loading array env", zap.String("key", s), zap.String("value", v), zap.Error(err))
```

Comment:
We should not log the error if we don't have an error.

```suggestion
			if err != nil {
				log.Error("config: error loading array env", zap.String("key", s), zap.String("value", v), zap.Error(err))
			}
```